ModLoader/ModLoader.h

```diff
@@ -0,0 +1,43 @@
+#ifndef MOD_LOADER_H
+#define MOD_LOADER_H
+#include <map>
+#include <optional>
+#include <string>
+#include <vector>
+
+using Mods = std::map<std::string, std::string>;							  // name, path.
+using IncomingMods = std::vector<std::pair<std::string, std::string>>; // vector of name/path pairs.
```

Comment:
It might be even clear if we introduce more aliases:

```
using Name = std::string;
using Path = std::string;
using Mods = std::map<Name, Path>;
using Mod = std::pair<Name, Path>;
using IncomingMods = std::vector<Mod>;
```

For that matter, Mod as suggested here might even be replaced with a struct so that the members are named instead of just ordered.